### PR TITLE
Split KeyboardLayoutPage & KeyboardLayoutModel

### PR DESCRIPTION
**Pull request by jpnurmi**
#147

### CODE REVIEW

File: packages/ubuntu_desktop_installer/lib/pages/keyboard_layout/keyboard_layout_model.dart [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import 'package:subiquity_client/subiquity_client.dart';
+
+import '../../keyboard_model.dart';
+
+/// Implements the business logic of the Keyboard Layout page.
+class KeyboardLayoutModel extends ChangeNotifier {
+  /// Creates a model with the specified [client] and [keyboardModel].
+  KeyboardLayoutModel({
+    required SubiquityClient client,
+    required KeyboardModel keyboardModel,
```
Comment (jpnurmi) [inline packages/ubuntu_desktop_installer/lib/pages/keyboard_layout/keyboard_layout_model.dart:12]: would be nice to rename KeyboardModel to avoid confusion between similarly named models.

Some ideas:
- KeyboardDatabase
- KeyboardRepository
- KeyboardService

Comment (oSoMoN) [reply packages/ubuntu_desktop_installer/lib/pages/keyboard_layout/keyboard_layout_model.dart:12]: Agreed. `KeyboardService` sounds fine to me.

**Comment by oSoMoN** [comment]
This is going to conflict with #154, so we'll need to be careful when rebasing.

**Comment by jpnurmi** [comment]
it's ok to merge the fix first. i can rebase later and add test cases for the new corner cases

**Comment by oSoMoN** [comment]
This now needs rebasing (thanks for reviewing my other branch that created this conflict!).

**Comment by jpnurmi** [comment]
It was easier to redo the changes than try to resolve the conflicts, but it's ok because the model changes are basically just moving small pieces of "logic" code from the page to the model. This time I did it in tiny steps, based on the latest main that contains the fix, to reduce the risk of regressions. Then I took the old PR tests and extended them with many cases with null/empty/unknown values.

File: packages/ubuntu_desktop_installer/lib/pages/keyboard_layout/keyboard_layout_model.dart (hunk 1 above)
Comment (jpnurmi) [reply packages/ubuntu_desktop_installer/lib/pages/keyboard_layout/keyboard_layout_model.dart:14]: :+1: I'll push a separate PR later

**Comment by oSoMoN** [comment]
That looks very good, thanks for this thorough refactoring work!
